react-search-component/my-app/src/SearchBar.tsx

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+type SearchBarProps = {
+  OnType: (e: React.ChangeEvent<HTMLInputElement>) => void;
```

Comment:
I think your code would be a little bit cleaner if this were declared as a function with a string parameter instead of an event parameter.